SU2_CFD/src/driver_direct_multizone.cpp

```diff
@@ -0,0 +1,689 @@
+    for (unsigned short iSol = 0; iSol < MAX_SOLS; iSol++){
+      /*-- If the solver position iSol is enabled --*/
+      if (solver_container[iZone][INST_0][MESH_0][iSol] != NULL){
+        solver_container[iZone][INST_0][MESH_0][iSol]->UpdateSolution_BGS(geometry_container[iZone][INST_0][MESH_0],
+            config_container[iZone]);}
+    }
+  }
+
+  if (rank == MASTER_NODE) cout.setf(ios::scientific, ios::floatfield);
+
+  /*-----------------------------------------------------------------*/
+  /*-------------------- Output FSI history -------------------------*/
+  /*-----------------------------------------------------------------*/
+  if (fsi){
+    bool ZONE_FLOW=0, ZONE_FEA=1;
+    /*--- This is a hack to test it works. ---*/
+    for (iZone = 0; iZone < nZone; iZone++){
+      if (config_container[iZone]->GetKind_Solver() == FEM_ELASTICITY) ZONE_FEA = iZone;
+      if (config_container[iZone]->GetKind_Solver() == NAVIER_STOKES) ZONE_FLOW = iZone;
+    }
+    output->SpecialOutput_FSI(&FSIHist_file, geometry_container, solver_container,
+        config_container, integration_container, 0,
+        ZONE_FLOW, ZONE_FEA, false);
+  }
+
+  return Convergence;
+
+}
+
+void CMultizoneDriver::Update() {
+
+  unsigned short jZone, UpdateMesh;
+  bool DeformMesh = false;
+
+  /*--- For enabling a consistent restart, we need to update the mesh with the interface information that introduces displacements --*/
+  /*--- Loop over the number of zones (IZONE) ---*/
+  for (iZone = 0; iZone < nZone; iZone++){
+
+    UpdateMesh = 0;
+
+      /*--- Transfer from all the remaining zones (JZONE != IZONE)---*/
+      for (jZone = 0; jZone < nZone; jZone++){
+        /*--- The target zone is iZone ---*/
+        if (jZone != iZone){
+          DeformMesh = Transfer_Data(jZone, iZone);
+          if (DeformMesh) UpdateMesh += 1;
+        }
+      }
+    /*--- If a mesh update is required due to the transfer of data ---*/
+    if (UpdateMesh > 0) DynamicMeshUpdate(iZone, ExtIter);
+
+    iteration_container[iZone][INST_0]->Update(output, integration_container, geometry_container,
+        solver_container, numerics_container, config_container,
+        surface_movement, grid_movement, FFDBox, iZone, INST_0);
+
+    /*--- Set the Convergence_FSI boolean to false for the next time step ---*/
+    for (unsigned short iSol = 0; iSol < MAX_SOLS; iSol++){
+      if (solver_container[iZone][INST_0][MESH_0][iSol] != NULL)
+        integration_container[iZone][INST_0][iSol]->SetConvergence_FSI(false);
+    }
+  }
+
+}
+
+void CMultizoneDriver::Output(unsigned long TimeIter) {
+
+  unsigned long nExtIter = config_container[ZONE_0]->GetnExtIter();
+  bool output_files = false;
+
+  /*--- Determine whether a solution needs to be written
+   after the current iteration ---*/
+
+  if (
+
+      /*--- General if statements to print output statements ---*/
+
+      (TimeIter+1 >= config_container[ZONE_0]->GetnTime_Iter()) || (StopCalc) ||
```

Comment:
I don't like it either. The idea is to move the conditional to the new output structure, where the logic is checked internally and case-dependent and only a method is called, so that's why I haven't really got into that yet.